gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/bridge/operation/ListEnvironmentOperationHandler.java

```diff
@@ -17,80 +17,65 @@
     }
 
     @Override
     public Single<BridgeReply> handle(BridgeCommand bridgeCommand) {
-        BridgeMultiReply multiReply = new BridgeMultiReply();
-        multiReply.setCommandId(bridgeCommand.getId());
+        String organizationId = bridgeCommand.getPayload().organizationId();
         try {
-            final List<EnvironmentEntity> managedEnvironments =
-                this.environmentService.findByOrganization(bridgeCommand.getOrganizationId());
-            multiReply.setCommandStatus(CommandStatus.SUCCEEDED);
-            multiReply.setReplies(
-                managedEnvironments
+            List<BridgeReplyPayload.BridgeReplyContent> replyContents =
+                this.environmentService.findByOrganization(organizationId)
                     .stream()
                     .map(environmentEntity -> {
-                        BridgeSimpleReply simpleReply = new BridgeSimpleReply();
-                        simpleReply.setCommandId(bridgeCommand.getId());
-                        simpleReply.setCommandStatus(CommandStatus.SUCCEEDED);
-                        simpleReply.setOrganizationId(environmentEntity.getOrganizationId());
-                        simpleReply.setEnvironmentId(environmentEntity.getId());
-                        simpleReply.setInstallationId(installationService.get().getId());
+                        BridgeReplyPayload.BridgeReplyContent.BridgeReplyContentBuilder builder = BridgeReplyPayload.BridgeReplyContent
+                            .builder()
+                            .environmentId(environmentEntity.getId())
+                            .organizationId(environmentEntity.getOrganizationId())
+                            .installationId(installationService.get().getId());
+
                         try {
-                            simpleReply.setPayload(objectMapper.writeValueAsString(environmentEntity));
+                            return BridgeReplyPayload.BridgeReplyContent
+                                .builder()
+                                .environmentId(environmentEntity.getId())
+                                .organizationId(environmentEntity.getOrganizationId())
+                                .installationId(installationService.get().getId())
+                                .content(objectMapper.writeValueAsString(environmentEntity))
+                                .build();
                         } catch (JsonProcessingException e) {
-                            logger.warn("Problem while serializing environment {}", environmentEntity.getId());
-                            simpleReply.setMessage("Problem while serializing environment: " + environmentEntity.getId());
-                            simpleReply.setCommandStatus(CommandStatus.ERROR);
+                            log.warn("Problem while serializing environment {}", environmentEntity.getId());
+                            return null;
```

Comment:
I'm not really familiar with Promotion, but is it intended not to send the error when a serialization problem occurs. Before we were sending an error reply 🤷